spi/crawler-spi/src/test/java/org/eclipse/edc/crawler/spi/model/RecurringExecutionPlanTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright (c) 2024 Fraunhofer-Gesellschaft
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer-Gesellschaft - initial implementation
+ *
+ */
+
+package org.eclipse.edc.crawler.spi.model;
+
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+public class RecurringExecutionPlanTest {
+    private static final Integer INTERVAL = 300;
+    private RecurringExecutionPlan recurringExecutionPlan;
+    private Monitor monitor;
+
+    @BeforeEach
+    public void setUp() {
+        monitor = mock(Monitor.class);
+        Duration schedule = Duration.ofMillis(100);
+        Duration initialDelay = Duration.ofMillis(50);
+        recurringExecutionPlan = new RecurringExecutionPlan(schedule, initialDelay, monitor);
+    }
+
+    @Test
+    public void runPlan_shouldExecuteAtLeastOnce() throws InterruptedException {
+        AtomicInteger counter = new AtomicInteger();
+        Runnable task = counter::incrementAndGet;
+
+        recurringExecutionPlan.run(task);
+        Thread.sleep(INTERVAL);
```

Comment:
We tend to avoid to have `Thread.sleep` in the tests, better to use `Awaitility` with the `pollDelay` set, e.g.:
```java
        await().pollDelay(20, TimeUnit.MILLISECONDS).untilAsserted(() -> {
            recurringExecutionPlan.stop();
            assertThat(counter).hasPositiveValue();
        });
```

durations, generally speaking should be as low as possible to keep execution fast, e.g.
plan initialDelay: 1ms
schedule: 5ms
assertion pollDelay: 10ms